pkg/controller/mongodb/mongodb_controller.go

```diff
@@ -291,24 +336,75 @@ func (r ReplicaSetReconciler) ensureAutomationConfig(mdb mdbv1.MongoDB) error {
 	return nil
 }
 
-func buildAutomationConfig(mdb mdbv1.MongoDB, mdbVersionConfig automationconfig.MongoDbVersionConfig, client mdbClient.Client) (automationconfig.AutomationConfig, error) {
-	domain := getDomain(mdb.ServiceName(), mdb.Namespace, "")
+// checkTLSConfig will check that the configured ConfigMap and Secret exist and that they have the correct fields.
+func (r *ReplicaSetReconciler) checkTLSConfig(mdb mdbv1.MongoDB) (bool, error) {
+	if !mdb.Spec.TLS.Enabled {
+		return false, nil
+	}
 
-	currentAc, err := getCurrentAutomationConfig(client, mdb)
-	if err != nil {
-		return automationconfig.AutomationConfig{}, err
+	r.log.Info("Ensuring TLS is correctly configured")
+
+	// Ensure CA ConfigMap exists
+	var caConfigMap corev1.ConfigMap
+	configMapName := types.NamespacedName{Name: mdb.Spec.TLS.CAConfigMapRef, Namespace: mdb.Namespace}
+	if err := r.client.Get(context.TODO(), configMapName, &caConfigMap); err != nil {
+		return errors.IsNotFound(err), err
 	}
-	newAc, err := automationconfig.NewBuilder().
+
+	// Ensure ConfigMap has a "ca.crt" field
+	if cert, ok := caConfigMap.Data[tlsCACertName]; !ok || cert == "" {
+		return true, fmt.Errorf(`ConfigMap "%s" should have a CA certificate in field "%s"`, caConfigMap.Name, tlsCACertName)
+	}
+
+	// Ensure Secret exists
+	var secret corev1.Secret
+	secretName := types.NamespacedName{Name: mdb.Spec.TLS.ServerSecretRef, Namespace: mdb.Namespace}
+	if err := r.client.Get(context.TODO(), secretName, &secret); err != nil {
+		return errors.IsNotFound(err), err
+	}
+
+	// Ensure Secret has "tls.crt" and "tls.key" fields
+	if key, ok := secret.Data[tlsSecretKeyName]; !ok || len(key) == 0 {
```

Comment:
again we have some functions in the `secret` package that will handle the common operations (or add some more if they don't meet your exact use case!)

a `secret.RequireKeys(secretGetter Getter, keys ...string) error` sounds like it could be a good fit